projects/packages/jetpack-mu-wpcom/src/features/blog-privacy/blog-privacy.php

```diff
@@ -18,12 +18,16 @@
 /**
  * Filters the robots.txt contents based on the value of the wpcom_data_sharing_opt_out option.
  *
- * @param string     $output The contents of robots.txt.
- * @param int|string $public The value of the blog_public option.
+ * @param string $output The contents of robots.txt.
+ * @param bool   $public Whether the site is considered public.
  * @return string
  */
-function robots_txt( string $output, $public ): string {
-	$public = (int) $public;
+function robots_txt( $output, $public ): string {
+	if ( ! is_string( $output ) ) {
+		$output = '';
+	}
+
+	$public = (int) get_option( 'blog_public' );
```

Comment:
Looking at this again, I think this could use a comment
```suggestion
	// WordPress passes in a bool. Re-fetch as an int to handle our custom "-1" value.
	$public = (int) get_option( 'blog_public' );
```